packages/react-link/src/components/Link/Link.tsx

```diff
@@ -5,13 +5,9 @@ import { renderLink } from './renderLink';
 import type { LinkProps } from './Link.types';
 
 /**
- * Defines a styled Link, using the `useLink` hook.
- * {@docCategory Link }
+ * A Link is a reference to data that a user can follow by clicking or tapping it.
  */
-export const Link: React.FunctionComponent<LinkProps & React.RefAttributes<HTMLElement>> = React.forwardRef<
-  HTMLElement,
-  LinkProps
->((props, ref) => {
+export const Link = React.forwardRef<HTMLElement, LinkProps>((props, ref) => {
```

Comment:
Please use proper typings if you would like to have this assignment.

```tsx
export const Link: React.ForwardRefExoticComponent<LinkProps> = React.forwardRef((props, ref) => {
```

---

### `ref` gets `unknown`
![image](https://user-images.githubusercontent.com/14183168/132823664-e2b84524-22c0-411e-92b2-3a8d74421627.png)

### `ref` is not a prop

- This highlights a bigger issue, we have `ref` in `LinkProps`, technically it's wrong as `ref` is **not** a prop in React

> That’s because ref is not a prop. Like key, it’s handled differently by React.
> https://reactjs.org/docs/forwarding-refs.html
```tsx
const props: LinkProps = {
  ref: React.createRef<HTMLButtonElement>()
}
```

IMO, we **should not** have `ref` in `*Props`

### `forwardRef()` returns different interface

`forwardRef()` returns `ForwardRefExoticComponent`, it's a different type and returned result is different: `FunctionComponent` represents functions, `ForwardRefExoticComponent` is **an object**

Reply:
Let's not block this PR on this since there is further discussion happening in #19812.